app/serializers/api/product_serializer.rb

```diff
@@ -16,16 +16,30 @@ class Api::ProductSerializer < ActiveModel::Serializer
   has_many :images, serializer: Api::ImageSerializer
   has_one :supplier, serializer: Api::IdSerializer
 
+  ALLOWED_CHARACTERS = {
+    "&amp;" => "&",
+    "&nbsp;" => " "
+  }.freeze
+
   # return an unformatted descripton
   def description
-    strip_tags object.description&.strip
+    return unless d = strip_tags(object.description&.strip)
+
+    ALLOWED_CHARACTERS.each do |character, sub|
+      d = d.gsub(character, sub)
+    end
+    d
   end
 
   # return a sanitized html description
   def description_html
     d = sanitize(object.description, tags: ["p", "b", "strong", "em", "i", "a", "u"],
                                      attributes: ["href", "target"])
-    d.to_s.html_safe
+    d = d.to_s.html_safe
+    ALLOWED_CHARACTERS.each do |character, sub|
+      d = d.gsub(character, sub)
```

Comment:
There's quite a bit of logic in this Serializer now for sanitizing HTML. It feels like it shouldn't be this class's responsibility. Maybe we could pull these bits into a small service object?

Reply:
03ede57b32f8066ee7bf6109145a47172d180eeb